### ISSUE TITLE
An alternative implementation

### ISSUE THREAD

**Issue by gdamore** [Decodetalkers/tree-sitter-meson#4]
Please see https://github.com/staysail/tree-sitter-meson

I believe that implementation is a more accurate grammar, and it does not suffer from the flaws identified here.  Additionally, a rich corpus of tests are included.

**Comment by Decodetalkers** [comment]
> Please see https://github.com/staysail/tree-sitter-meson
> 
> I believe that implementation is a more accurate grammar, and it does not suffer from the flaws identified here.  Additionally, a rich corpus of tests are included.
> 

Yes, it is a pretty bad grammar, because I just want the highlight work, so if you gammar is better, then you can make a pull request to neovim, I am glad that a better one is created, and if you need help, I will try.

**Comment by gdamore** [comment]
I don't use neovim (could never figure out the plugin syntax etc.).

I'm happy to have this integrated elsewhere though.  I've posted a PR to update Helix to use it, and I also have a Nova extension that uses it on the market. 

If I can figure it out I'll see what I can do.
